prooph/psb-zeromq-producer
# Initial release.

#### What's this PR do?

This allows the Prooph Service Bus to use ZeroMQ to fire commands. ZeroMQ does not provide a query bus implementation but in the future that is something we could look at with the zero-rpc wrapper.
#### Where should the reviewer start?

You will need a ZeroMQ server and a PHP installation with `ext-zmq`. There will soon be a Dockerfile which will pull the dependencies and run the tests.
#### How should this be manually tested?
1. Setup the ZeroMQ server.
2. Add a consumer to the ZMQ server.
3. Set the PSB adapter to use `Prooph\ServiceBus\Message\ZeroMQ\ZeroMQMessageProducer`
4. Configure the ZMQ adapter, by adding the DSN required. (more doc later)
5. Dispatch a command from the PSB.
6. Check that the consumer gets the command.
#### Any background context you want to provide?

There was an outstanding PSB requirement for the ZMQ adapter.

There are a few other implementations which inspired this implementation.
- [psb-http-producer](https://github.com/prooph/psb-http-producer)
- [psb-bernard-producer](https://github.com/prooph/psb-bernard-producer)
#### Definition of Done:
- [x] Tests are all passing.
- [x] Been tested with a live ZMQ server and consumer.
- [x] Dockerfile to run tests with all dependencies installed.
- [x] Travis-CI is able to run tests. (I believe it can use the Dockerfile)
